src/main/java/com/easysplit/ess/transactions/infrastructure/persistence/TransactionsRepositoryImpl.java

```diff
@@ -160,4 +164,72 @@ public int getLastRevision(String debtGuid) {
 
         return revision;
     }
+
+    @Override
+    public TransactionEntity getTransaction(String transactionGuid){
+        TransactionEntity transactionEntity = null;
+
+        try{
+            transactionEntity = jdbc.query(TransactionsQueries.GET_TRANSACTION,
+                    this::toTransactionEntity,
+                    transactionGuid);
+        }catch(NotFoundException e){
+            //Catching NotFoundException thrown from toTransactionEntity method
+            logger.debug(CLASS_NAME + ".getTransaction() - NotFoundException while reading the transaction: " + transactionGuid, e);
+            throw e;
+        }catch(Exception e){
+            logger.error(CLASS_NAME + ".getTransaction() - Something went wrong while reading the group with id: "+ transactionGuid, e);
+            infrastructureHelper.throwInternalServerErrorException(
+                    ErrorKeys.GET_TRANSACTION_ERROR_TITLE,
+                    ErrorKeys.GET_TRANSACTION_ERROR_MESSAGE,
+                    new Object[] {transactionGuid},
+                    e
+            );
+        }
+
+        return transactionEntity;
+    }
+
+    private TransactionEntity toTransactionEntity(ResultSet rs) throws SQLException {
+
+        TransactionEntity transactionEntity = null;
+
+        if(rs.next()) {
+            transactionEntity = new TransactionEntity();
+
+            transactionEntity.setTransactionGuid(rs.getString(TransactionsQueries.TRANSACTIONGUID_COLUMN.toLowerCase()));
+            transactionEntity.setName(rs.getString(TransactionsQueries.NAME_COLUMN.toLowerCase()));
+
+            String creditorByGuid = rs.getString(TransactionsQueries.CREDITOR_COLUMN.toLowerCase());
+            UserEntity creditor = userRepository.getUser(creditorByGuid);
+            transactionEntity.setCreditor(creditor);
+
+            String debtorByGuid = rs.getString(TransactionsQueries.DEBTOR_COLUMN.toLowerCase());
+            UserEntity debtor = userRepository.getUser(debtorByGuid);
+            transactionEntity.setDebtor(debtor);
+
+            //DEBT DETAILS HERE
```

Comment:
I dont think we should set the debt details in here, instead we should use the method I mentioned we are missing in the TransactionServiceImpl.getTransaction()